xml/tuning-os-tuning.xml

```diff
@@ -0,0 +1,425 @@
+     Numerous other references were utilized, including documentation
+     from hardware vendors.
+   </para>
+   <sect2>
+     <title>CPU Mitigations</title>
+     <para>
+       One key area of kernel for performance tuning is to disable the
+       side-channel attack mitigations present in the kernel. The
+       bulk of the benefits from this tuning occur with smaller I/Os
+       ranging in size from 4k to 64k. In particular, 64K random read
+       and sequential writes doubled in performance in a limited test
+       environment using only two client nodes.
+     </para>
+     <para>
+       Changing these options requires a clear understanding of the security
+       implications as they involve disabling mitigations for side-chanel
+       attacks on some CPUs. The need to disable these mitigations may
+       be minimized with newer processors. Evaluate carefully whether
+       this is something needed for the particular hardware being utilized.
+       In the test configuration, a salt state was utilized to apply these
+       changes. The salt state should be in a subdirectory of <filename>/srv/salt</filename>
+       on the &smaster; and is applied by using a <command>salt state.apply</command> command
+       similar to below:
+     </para>
+<screen>
+salt '*' state.apply my_kerntune
+</screen>
+     <para>
+       The salt state and steps used in this testing can be found
+       in Appendix A. This needs to be adjusted to work in each
+       customer environment.
+     </para>
+   </sect2>
+   <sect2>
+     <title>I/O tuning - Multiqueue Block I/O</title>
+     <para>
+       The first aspect to tune is to ensure that I/O if flowing in the
+       most optimal pattern. For the test cluster used in this test,
+       that means enable multi-queue block I/O. This is done through
+       adding a boot-time kernel parameter as found in the Section 12.4
+       <link xlink:href="https://documentation.suse.com/sles/15-SP1/html/SLES-all/cha-tuning-io.html#cha-tuning-io-barrier"/>.
+       This is not an action that should be taken unilaterally on clusters
+       that contain spinning media devices due to potential performance
+       degradation for those devices. The general result is that there are
+       multiple I/O queues assigned to the device, allowing more jobs to be
+       handled simultaneously by those high performance device such as NVMe
+       and SSD that can service large numbers of requests.
+     </para>
+   </sect2>
+   <sect2>
+     <title>SSD Tuning</title>
+     <para>
+       To get the best read performance, it may be necessary to adjust
+       the <option>read_ahead</option> and write cache settings for the SSD devices.
+       In our particular testing environment, disabling write cache
+       and forcing <option>read_ahead</option> to 2MB resulted in the best
+       overall performance.
+       Before tuning, it is important to check the default values and
```

Comment:
Odd line-break here. Should this be in one paragraph, or split into 2?

Reply:
Ah yeah, I've done this once or twice when I should have broken the <para> tags up